src/include/platform/NetworkCommissioning.h

```diff
@@ -357,7 +357,12 @@ class WiFiDriver : public Internal::WirelessDriver
      *  Provide a default implementation that returns the 2.4 Ghz band support.
      *  Note: WiFi platforms should implement this function in their WiFiDriver to provide their complete device capabilities
      */
-    virtual WiFiBand GetSupportedWiFiBands() { return WiFiBand::k2g4; }
+    virtual Span<const WiFiBand> GetSupportedWiFiBands()
```

Comment:
So this API shape requires the callee to always have the entire list in RAM, right?

Maybe that's OK because this will generally be a pretty small list, but just checking.  It would be good to document that restriction in the API definition, because it might not be at all obvious to someone implementing this API that the buffer they return has to live forever.

Also, I guess this is not a compat break because we have not shipped this yet, right?

Reply:
Right, this is not a compat break. Also, I changed the API to be more consistent with the rest of the codebase after seeing your comment. It's true that returning a span is more error-prone that using a caller-initialized span.